docker_mounts.go

```diff
@@ -85,6 +100,40 @@ func (s DockerTmpfsMountSource) GetTmpfsOptions() *mount.TmpfsOptions {
 	return s.TmpfsOptions
 }
 
+// DockerImageMountSource is a mount source for an image
+type DockerImageMountSource struct {
+	// ImageName is the image name
+	ImageName string
```

Comment:
question: do these need to be public so should we keep them private and use a NewXXX method to ensure the type is constructed properly?